src/secondary.rs

```diff
@@ -689,6 +689,41 @@ impl<K: Key, V> SecondaryMap<K, V> {
             inner: self.iter_mut(),
         }
     }
+
+    /// Gets the given key's corresponding entry in the map for in-place manipulation.
+    /// Returns None if the key is invalid.
```

Comment:
I would relax this to 'may return `None`' or something more specific that signifies awareness that having a key be invalid in the main map does not automagically make it detectable as invalid in the secondary map.